python/lsst/sims/featureScheduler/basis_functions/mask_basis_funcs.py

```diff
@@ -0,0 +1,206 @@
+import numpy as np
+import healpy as hp
+from lsst.sims.utils import _hpid2RaDec, Site, _angularSeparation
+import matplotlib.pylab as plt
+from lsst.sims.featureScheduler.basis_functions import Base_basis_function
+
+
+__all__ = ['Zenith_mask_basis_function', 'Zenith_shadow_mask_basis_function',
+           'Moon_avoidance_basis_function', 'Map_cloud_basis_function']
+
+
+class Zenith_mask_basis_function(Base_basis_function):
+    """Just remove the area near zenith.
+
+    Parameters
+    ----------
+    min_alt : float (20.)
+        The minimum possible altitude (degrees)
+    max_alt : float (82.)
+        The maximum allowed altitude (degrees)
+    """
+    def __init__(self, min_alt=20., max_alt=82.):
+        super(Zenith_mask_basis_function, self).__init__()
+        self.update_on_newobs = False
+        self.min_alt = np.radians(min_alt)
+        self.max_alt = np.radians(max_alt)
+        self.result = np.empty(hp.nside2npix(self.nside), dtype=float).fill(self.penalty)
+
+    def _calc_value(self, conditions, indx=None):
+
+        result = self.result.copy()
+        alt_limit = np.where((conditions.alt > self.min_alt) &
+                             (conditions.alt < self.max_alt))[0]
+        result[alt_limit] = 1
+        return result
+
+
+class Zenith_shadow_mask_basis_function(Base_basis_function):
+    """Mask the zenith, and things that will soon pass near zenith. Useful for making sure
+    observations will be able to be too close to zenith when they need to be observed again (e.g. for a pair)
```

Comment:
'will be able to be too close' -> 'will not be too close'? or am i misunderstanding this?